packages/docusaurus/src/server/config.ts

```diff
@@ -56,39 +21,5 @@ export default function loadConfig(siteDir: string): DocusaurusConfig {
   }
 
   const loadedConfig = importFresh(configPath) as Partial<DocusaurusConfig>;
-  const missingFields = REQUIRED_FIELDS.filter(
-    (field) => !has(loadedConfig, field),
-  );
-
-  if (missingFields.length > 0) {
-    throw new Error(
-      `The required field(s) ${formatFields(
-        missingFields,
-      )} are missing from ${CONFIG_FILE_NAME}`,
-    );
-  }
-
-  // Merge default config with loaded config.
-  const config: DocusaurusConfig = {
-    ...DEFAULT_CONFIG,
-    ...loadedConfig,
-  } as DocusaurusConfig;
-
-  // Don't allow unrecognized fields.
-  const allowedFields = [...REQUIRED_FIELDS, ...OPTIONAL_FIELDS];
-  const unrecognizedFields = Object.keys(config).filter(
-    (field) => !allowedFields.includes(field),
-  );
-
-  if (unrecognizedFields && unrecognizedFields.length > 0) {
-    throw new Error(
-      `The field(s) ${formatFields(
-        unrecognizedFields,
-      )} are not recognized in ${CONFIG_FILE_NAME}.
-If you still want these fields to be in your configuration, put them in the 'customFields' attribute.
-See https://v2.docusaurus.io/docs/docusaurus.config.js/#customfields`,
-    );
-  }
-
-  return config;
+  return Joi.attempt(loadedConfig, ConfigSchema);
```

Comment:
here it would be better to abortEarly: false

so you need to catch the potential validationError, extract all the errors, and rethrow as a well-formatted regular error (with line breaks etc...)